test/integration/pkg/generated/openapi_generated.go

```diff
@@ -67,6 +69,80 @@ func schema_test_integration_testdata_custom_Bah(ref common.ReferenceCallback) c
 	})
 }
 
+func schema_test_integration_testdata_defaults_Defaulted(ref common.ReferenceCallback) common.OpenAPIDefinition {
+	return common.OpenAPIDefinition{
+		Schema: spec.Schema{
+			SchemaProps: spec.SchemaProps{
+				Type: []string{"object"},
+				Properties: map[string]spec.Schema{
+					"Field": {
+						SchemaProps: spec.SchemaProps{
+							Default: "bar",
+							Type:    []string{"string"},
+							Format:  "",
+						},
+					},
+					"OtherField": {
+						SchemaProps: spec.SchemaProps{
+							Default: 0,
+							Type:    []string{"integer"},
+							Format:  "int32",
+						},
+					},
+					"list": {
+						SchemaProps: spec.SchemaProps{
+							Default: []interface{}{"foo", "bar"},
```

Comment:
Do we make sure that these are not published to clients?

Before vendoring this into kube this must be put in place.

Reply:
Curious why we wouldn't publish that to clients?

Reply:
To avoid bad behaviour in the ecosystem: like clients that take action on default values, e.g. default objects themselves instead of on the server.

Reply:
We can't prevent people from doing the wrong thing. I think the more description the API is the better.

Reply:
I don't agree. Information hiding is a good thing if you don't want to create inofficially used APIs.

Reply:
also I bet that OpenAPI compatible/generated clients will apply defaults. And our default are not what they think they are. They have non-standard semantics.

Reply:
I'm 100% sure that the clients will NOT apply these defaults because that's precisely AGAINST the definition of the field, which is merely to document what the server will do:

> The default value is the one that the server uses if the client does not supply the parameter value in the request.

We want to be able to describe programmatically how our APIs work.

Reply:
But yeah, I agree that we don't want clients to apply the defaults for us.

Reply:
@sttts can you ack this point? I don't want to merge if there's a pending disagreement, Antoine's answer sounds convincing to me.

Reply:
I ended-up removing the defaults as Stefan recommended. We'll revisit that holistically later.